File: src/main/java/com/wellsfargo/counselor/entity/portfolio.js
package com.wellsfargo.counselor.entity;

import javax.persistence.*;
import java.util.List;

@Entity
public class Portfolio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(mappedBy = "portfolio", cascade = CascadeType.ALL)
    private List<Asset> assets;

    @ManyToOne
    @JoinColumn(name = "client_id", nullable = false)
    private Client client;

    // Constructors, getters, and setters

    public Portfolio() {
    }

    public Portfolio(Client client, List<Asset> assets) {
        this.client = client;
        this.assets = assets;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public List<Asset> getAssets() {
        return assets;
    }

    public void setAssets(List<Asset> assets) {
        this.assets = assets;
    }

    public Client getClient() {
        return client;
    }

    public void setClient(Client client) {
        this.client = client;
    }
}

File: src/main/java/com/wellsfargo/counselor/entity/portfolio.java
package com.wellsfargo.counselor.entity;
import java.util.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;

@Entity
public class portfolio {
	
	@Id
	@GeneratedValue()
	private long portfolioId;
	
	@Id
	@Column(nullable = false)
	private long clientId;
	
	@Column(nullable = false)
	private Date creationDate;
	
	public portfolio(long clientId, Date creationDate) {
		this.clientId = clientId;
		this.creationDate = creationDate;
		
	}
	
	public long getPortfolioId() {
		return portfolioId;
	}
	
	public long getClientId() {
		return clientId;
	}
	
	public void setClientId(long id) {
		this.clientId = id;
	}
	
	public Date getCreationDate() {
		return creationDate;
	}
	
	public void setCreationDate(Date date) {
		this.creationDate = date;
	}
}
